### PR TITLE
feat: proxy protocol in ClientTrafficPolicy

### PR DESCRIPTION
**Pull request by arkodg**
relates to https://github.com/envoyproxy/gateway/issues/1328

### CODE REVIEW

**Review by shawnh2 — APPROVED** [review]
lgtm

**Comment by zirain** [comment]
lgtm, is it possible to add e2e tests for this?

File: internal/xds/translator/proxy_protocol.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import (
+	listenerv3 "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	proxyprotocolv3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/listener/proxy_protocol/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/wellknown"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/envoyproxy/gateway/internal/ir"
+)
+
+// patchProxyProtocolFilter builds and appends the Proxy Protocol Filter to the HTTP
```
Comment (zhaohuabing) [inline internal/xds/translator/proxy_protocol.go:17]: nit: patchProxyProtocolFilter builds and appends the Proxy Protocol Filter to the HTTP Listener if applicable.

File: internal/xds/translator/proxy_protocol.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	}
+
+	proxyProtocolFilter := buildProxyProtocolFilter()
+
+	if proxyProtocolFilter != nil {
+		xdsListener.ListenerFilters = append(xdsListener.ListenerFilters, proxyProtocolFilter)
+	}
+}
+
+// buildProxypProtocolFilter returns a Proxy Protocol filter from the provided IR listener.
```
Comment (zhaohuabing) [inline internal/xds/translator/proxy_protocol.go:39]: nit: buildProxypProtocolFilter returns a Proxy Protocol listener filter from the provided IR listener.

**Review by zhaohuabing — APPROVED** [review]

**Comment by arkodg** [comment]
> lgtm, is it possible to add e2e tests for this?

@zirain should be possible, can do in a follow up PR using https://github.com/pires/go-proxyproto

File: internal/xds/translator/proxy_protocol.go (hunk 1 above)
Comment (arkodg) [reply internal/xds/translator/proxy_protocol.go:17]: uggh my bad

**Comment by zirain** [comment]
I think this will confict with https://github.com/envoyproxy/gateway/pull/2210

**Review by zirain — APPROVED** [review]

**Pull request merged**